modules/extensions/src/collide/shader-module.ts

```diff
@@ -0,0 +1,130 @@
+import {project} from '@deck.gl/core';
+import type {_ShaderModule as ShaderModule} from '@deck.gl/core';
+import type {Texture2D} from '@luma.gl/webgl';
+
+const vs = `
+#ifdef NON_INSTANCED_MODEL
+attribute float collidePriorities;
+#else
+attribute float instanceCollidePriorities;
+#endif
+
+uniform bool collide_sort;
+
+vec2 collide_getCoords(vec4 position) {
+  vec4 collide_clipspace = project_common_position_to_clipspace(position);
+  return (1.0 + collide_clipspace.xy / collide_clipspace.w) / 2.0;
+}
+`;
+
+const fs = `
+uniform sampler2D collide_texture;
+uniform bool collide_enabled;
+
+float match(vec2 tex, vec3 pickingColor) {
+  vec4 collide_pickingColor = texture2D(collide_texture, tex);
+  float delta = distance(collide_pickingColor.rgb, pickingColor);
+  float e = 0.000001;
+  return step(delta, e);
+}
+
+float collide_isInBounds(vec2 texCoords, vec3 pickingColor) {
+  if (!collide_enabled) {
+    return 1.0;
+  }
+
+  float O = 0.0;
+  vec2 dx = 2.0 * vec2(1.0 / 2048.0, 0.0);
+  vec2 dy = dx.yx;
+
+  // Visibility test
```

Comment:
- Use for loops?
- Is it necessary to sample so many pixels? Add some comments?

Reply:
Switched to for-loop. I'm not 100% happy with this solution, as the object flicker when the view is tilted as the aliasing produces slightly different results between frame. A better solution might be to add some form of leaky bucket to smooth these high-frequency changes out over a few frames, but I'm not going to attempt that in this PR